lib/config/configuration.go

```diff
@@ -1570,9 +1570,10 @@ func applyDiscoveryConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 		}
 
 		var assumeRole *types.AssumeRole
-		if matcher.AssumeRoleARN != "" || matcher.ExternalID != "" {
+		if matcher.AssumeRoleARN != "" || matcher.ExternalID != "" || matcher.AssumeRoleName != "" {
```

Comment:
That `if` isn't actually needed, right? We can just always assign that field?

Reply:
If those are all empty, we get a non-nil `types.AssumeRole` where previously we would get a nil object.

It's a subtle change but there's nil vs non-nil checks afterwards in other parts of the code, so I would rather keep that as is.